api/src/test/java/org/apache/iceberg/expressions/TestStringLiteralConversions.java

```diff
@@ -181,17 +226,30 @@ public void testNegativeStringToTimestampLiteral() {
   @Test
   public void testTimestampWithZoneWithoutZoneInLiteral() {
     // Zone must be present in literals when converting to timestamp with zone
-    Literal<CharSequence> timestampStr = Literal.of("2017-08-18T14:21:01.919");
-    assertThatThrownBy(() -> timestampStr.to(Types.TimestampType.withZone()))
+    assertThatThrownBy(
+            () -> Literal.of("2017-08-18T14:21:01.919123").to(Types.TimestampType.withZone()))
```

Comment:
any reason this changed from `2017-08-18T14:21:01.919`?

Reply:
This wasn't a technically necessary change, but it does make human-readability slightly easier by matching "microseconds" with six digits, and "nanoseconds" with nine digits. I made this change in several places, all for the same reason.